tests/integration_tests/sqla_models_tests.py

```diff
@@ -164,6 +164,24 @@ def test_extra_cache_keys(self, mock_user_email, mock_username, mock_user_id):
         self.assertTrue(table1.has_extra_cache_key_calls(query_obj))
         assert set(extra_cache_keys) == {1, "abc", "abc@test.com"}
 
+        # Table with Jinja callable with set.
+        table1 = SqlaTable(
+            table_name="test_has_extra_cache_keys_table_with_set",
+            sql="""
+            {% set user_email = current_user_email() %}
+            SELECT
+              '{{ current_user_id() }}' as id,
+              '{{ current_username() }}' as username,
+              '{{ user_email }}' as email
+            """,
+            database=get_example_database(),
+        )
+
+        query_obj = dict(**base_query_obj, extras={})
+        extra_cache_keys = table1.get_extra_cache_keys(query_obj)
+        assert table1.has_extra_cache_key_calls(query_obj) is True
+        assert set(extra_cache_keys) == {1, "abc", "abc@test.com"}
+
```

Comment:
bycatch - this test is huge, and could do with some heavy parametrization. Not necessarily something that has to be done here, but it would leave the codebase in a much tidier state.